daemon/restart.go

```diff
@@ -35,10 +35,19 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		defer daemon.Unmount(container)
 	}
 
+	// set AutoRemove flag to false before stop so the container won't be
+	// removed during restart process
+	autoRemove := container.HostConfig.AutoRemove
+	defer func() {
```

Comment:
Let me know if you think I'm being paranoid, but I'd prefer this:
```
    // set AutoRemove flag to false before stop so the container won't be
    // removed during restart process
    autoRemove := container.HostConfig.AutoRemove

    container.HostConfig.AutoRemove = false
    err := daemon.containerStop(container, seconds)
    container.HostConfig.AutoRemove = autoRemove

    if err != nil {
        return err
    }

    if err := daemon.containerStart(container); err != nil {
        return err
    }
```

because I'm worried about the defer resetting the container.HostConfig.AutoRemove when perhaps it shouldn't. For example, normally I wouldn't expect containerStart() to change that value but you never know what might happen in the future. So to prevent unexpected side-effects of the defer, I think it would be safe to just clear and then reset AutoRemove right around the containerStop() since that's really the only code we need to be worried about.